topic-configuration/README.adoc

```diff
@@ -0,0 +1,689 @@
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Integer
+
+h|Default value
+|`2`
+
+h|Kafka property name
+|`min.insync.replicas`
+|===
+--
+
+Retention time::
++
+--
+Retention time is the amount of time that messages are retained in a topic before they are deleted. This property applies only when the topic cleanup policy is set to `Delete` or `Compact, Delete`.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|Yes
+
+h|Type
+|Long
+
+h|Default value
+|`604800000 ms` (7 days)
+
+h|Supported values
+| `milliseconds`, `seconds`, `days`, `months`, `years`, `Unlimited`
+
+h|Kafka property name
+|`retention.ms`
+|===
+--
+
+Retention size::
++
+--
+Retention size is the maximum total size of all log segments in a partition before old log segments are deleted to free up space. By default, no retention size limit is applied, only a retention time limit. This property applies only when the topic cleanup policy is set to `Delete` or `Compact, Delete`.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|Yes
+
+h|Type
+|Long
+
+h|Default value
+|`Unlimited`
+
+h|Supported values
+| `bytes`, `kilobytes`, `megabytes`, `gigabytes`, `terabytes`, `Unlimited`
+
+h|Kafka property name
+|`retention.bytes`
+|===
+--
+
+=== Messages
+
+These properties control how your messages are handled in the Kafka instance.
+
+Maximum message bytes::
++
+--
+Maximum message bytes is the maximum record batch size.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Integer
+
+h|Default value
+|`1 MB`
+
+h|Kafka property name
+|`max.message.bytes`
+|===
+--
+
+Message timestamp type::
++
+--
+Message timestamp type determines whether the timestamp is generated when the message is created (`CreateTime`) or when the message is appended to the log (`LogAppendTime`).
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|String
+
+h|Default value
+|`CreateTime`
+
+h|Kafka property name
+|`message.timestamp.type`
+|===
+--
+
+Maximum message timestamp difference::
++
+--
+Maximum message timestamp difference is the maximum difference allowed between the timestamp specified in the message when it leaves the producer and the timestamp recorded when a broker receives the message.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Long
+
+h|Default value
+|`9223372036854775807 ms` (292271023 years, 2 weeks, 2 days, 13 hours, 12 minutes, 56.064 seconds)
+
+h|Kafka property name
+|`message.timestamp.difference.max.ms`
+|===
+--
+
+Compression type::
++
+--
+Compression type determines the final compression for the topic. The `Producer` compression type retains the original compression set by the producer.
```

Comment:
Say anything on the compression types the producer can use?

Reply:
Not sure I follow. The compression type is set to either `Producer` or `Consumer`, but in our product defaults to only `Producer` and isn't changeable.

Reply:
Yes. Not clear. I was wondering if we wanted to say anything about supported compression types for the producer (gzip, snappy, zstd)

Reply:
@Simon, what do you think? ^^

Reply:
I think Paul's suggestion of listing the compression types is a good one

Reply:
Done. Reworded to this:

>Compression type determines the final compression for the topic. The standard compression types are `gzip`, `snappy`, `lz4`, and `zstd`. Additional compression types include `Uncompressed`, which does not compress the topic, and `Producer`, which retains the original compression type set by the producer. By default, the compression type is set to `Producer`.

Good with you Simon?

Reply:
LGTM